enhancements/ingress/configurable-dns-loadbalancerservice.md

```diff
@@ -0,0 +1,343 @@
+}
+
+// LoadBalancerDNSManagementPolicy is a policy for configuring how
+// ingresscontrollers manage DNS.
+//
+// +kubebuilder:validation:Enum=Managed;Unmanaged
+type LoadBalancerDNSManagementPolicy string
+
+const (
+    // ManagedLoadBalancerDNS configures the ingresscontroller to manage
+    // the wildcard DNS records associated with it.
+    ManagedLoadBalancerDNS LoadBalancerDNSManagementPolicy = "Managed"
+    // UnmanagedLoadBalancerDNS configures the ingresscontroller to not
+    // manage the wildcard DNS records associated with it.
+    UnmanagedLoadBalancerDNS LoadBalancerDNSManagementPolicy = "Unmanaged"
+)
+
+// DNSRecordSpec contains the details of a DNS record.
+type DNSRecordSpec struct {
+  // <snip>
+
+  // dnsManagementPolicy denotes the current policy applied on the dns record.
+  // Records that have policy set as "Unmanaged" are ignored by the ingress
+  // operator and the DNS record on the cloud provider is not managed by the
+  // operator. This record on the cloud provider can be deleted at the
+  // discretion of the cluster admin.
+  //
+  // +kubebuilder:default:="Managed"
+  // +kubebuilder:validation:Optional
+  // +kubebuilder:validation:Enum=Managed;Unmanaged
+  DNSManagementPolicy DNSManagementPolicy `json:"dnsManagementPolicy"`
+}
+
+// DNSManagementPolicy is a policy for configuring how the dns controller 
+// manages DNSRecords.
+//
+// +kubebuilder:validation:Enum=Managed;Unmanaged
+type DNSManagementPolicy string
+
+const (
+    // ManagedDNS configures the dns controller manage the lifecyle of the
+    // dns record on the appropriate platform.
+    ManagedDNS DNSManagementPolicy = "Managed"
+    // UnmanagedDNS configures the dns controller to ignore the dns record
+    // and allows the DNS record on the cloud provider to be managed
+    // by the cluster admin.
+    UnmanagedDNS DNSManagementPolicy = "Unmanaged"
+)
+```
+
+Re-uses existing condition `DNSManaged` of the controller to denote if
+the wildcard *DNSRecord* associated with the controller is managed/unmanaged. 
+
+### Implementation Details/Notes/Constraints
+
+Based on `.loadbalancer.dnsManagementPolicy`, the controller decides whether to ensure creating
+a wildcard *DNSRecord*. If the customer is updating `.loadbalancer.dnsManagementPolicy` from 
+`"Managed"` to `"Unmanaged"` the previously created *DNSRecord* will be updated
+appropriately.
+
+The existing `.loadbalancer.dnsManagementPolicy` value of the ingresscontroller
+will be replicated to the *DNSRecord*'s `.spec.dnsManagementPolicy` as well, to
+clearly state which dns record is unmanaged. The cluster admin can choose to delete
+the *DNSRecord* at his own discretion (this will involve also manually removing the finalizer).
+
+Once the *DNSRecord* is set to `Unmanaged`, any updates done to it will be ignored
+by the operator, but it is to be noted that if the ingresscontroller is updated from
+`Unmanaged` to `Managed` any changes done will be lost during re-sync.
```

Comment:
I needed to think about this to make sure there wasn't a race condition here:

1. `oc -n openshift-ingress-operator patch ingresscontrollers/default --type=merge --patch='{"spec":{"endpointPublishingStrategy":{"loadBalancer":{"dnsManagementPolicy":"Unmanaged"}}}}'`
2. `oc -n openshift-ingress patch dnsrecords/default-wildcard --type=merge --patch='{"spec":{"recordTTL":123}}'`
3. `oc -n openshift-ingress-operator patch ingresscontrollers/default --type=merge --patch='{"spec":{"endpointPublishingStrategy":{"loadBalancer":{"dnsManagementPolicy":"Managed"}}}}'`

We should be fine because the operator will patch the DNSRecord CR's `spec.recordTTL` field at the same time that the operator patches the CR's `spec.dnsManagementPolicy` field, so the dns controller will never will never observe the user-patched TTL at the same time that it observes "Managed".  

It shouldn't _really_ matter because users should _never_ patch the DNSRecord CR, but... you never know.